signac/contrib/project.py

```diff
@@ -1544,15 +1570,12 @@ def init_project(cls, name, root=None, workspace=None, make_dir=True):
             assert project.id == str(name)
             return project
         else:
-            if project.id != str(name) or (
-                workspace is not None
-                and os.path.realpath(workspace) != os.path.realpath(project.workspace())
-            ):
+            if workspace is not None and os.path.realpath(
+                workspace
+            ) != os.path.realpath(project.workspace()):
```

Comment:
**Not in this PR:** We should aim to consistently use the `pathlib.Path` API for all `os.PathLike` objects.

Reply:
In this specific case, I think realpath is the best choice — it is necessary and sufficient for the check we want to perform.